Backend/TimeTracker.Api/Controllers/TimerController.cs

```diff
@@ -0,0 +1,206 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using TimeTracker.Api.Database;
+using TimeTracker.Api.Database.Models;
+using TimeTracker.Api.DTOs;
+using TimeTracker.Api.Helpers;
+
+namespace Backend.Controllers
+{
+    [ApiController]
+    [Authorize]
+    [Route("[controller]")]
+    public class TimerController : ControllerBase {
+        MainDb database;
+        AuthHelper authHelper;
+
+        public TimerController(MainDb db, AuthHelper auth)
+        {
+            database = db;
+            authHelper = auth;
+        }
+
+        // returns a TimerDTO for the newly created Timer
+        [HttpPost]
+        [Route("Start")]
+        public async Task<GenericResponseDTO<TimerDTO>> StartTimer(TimerCreateDTO timerInfo) {
+
+            if(User == null) {
+                return new GenericResponseDTO<TimerDTO> {
+                    Message = "Not logged in.",
+                    Success = false
+                };
+            }
+
+            Project project = await database.Projects
+                .FirstOrDefaultAsync(p => p.Id == timerInfo.ProjectId);
+
+            if(project == null) {
+                return new GenericResponseDTO<TimerDTO> {
+                    Message = "Could not find project.",
+                    Success = false
+                };
+            }
+
+            Timer timer = (await database.Timers
+                .AddAsync(new Timer {
+                    User = await authHelper.GetCurrentUser(User, database),
+                    StartTime = DateTime.Now,
+                    Notes = timerInfo.Notes,
+                    Project = project
+                })).Entity;
+
+            await database.SaveChangesAsync();
+
+            return new GenericResponseDTO<TimerDTO> {
+                Data = new TimerDTO {
+                    Id = timer.Id,
+                    StartTime = timer.StartTime,
+                    Notes = timer.Notes,
+                    ProjectId = timer.Project.Id
+                },
+                Success = true
+            };
+            
+        }
+
+        // returns a TimeEntryDTO for the newly created TimeEntry
+        [HttpPost]
+        [Route("Stop")]
+        public async Task<GenericResponseDTO<TimeEntryDTO>> StopTimer(int timerId) {
+
+            if(User == null) {
+                return new GenericResponseDTO<TimeEntryDTO> {
+                    Message = "Not logged in.",
+                    Success = false
+                };
+            }
+
+            Timer timer = await database.Timers
+                .Include(t => t.User)
+                .Include(t => t.Project)
+                .FirstOrDefaultAsync(t => t.Id == timerId);
+
+            if(timer.User.Id != authHelper.GetCurrentUserId(User)) {
+                return new GenericResponseDTO<TimeEntryDTO> {
+                    Message = "Invalid user.",
+                    Success = false
+                };
+            }
```

Comment:
See my comment a little further down. Remove include user and do the user check in the database call.